src/nicypkg/functions.nim

```diff
@@ -99,35 +92,33 @@ proc gitBranch*(): string =
   let (o, err) = execCmdEx("git status")
   if err == 0:
     let firstLine = o.split("\n")[0].split(" ")
-    result = firstLine[firstLine.len - 1] & " "
+    result = firstLine[^1] & " "
   else:
     result = ""
 
 proc gitStatus*(dirty, clean: string): string =
   let (o, err) = execCmdEx("git status --porcelain")
-  if err == 0:
+  result = if err == 0:
     if o.len != 0:
-      result = dirty
+      dirty
     else:
-      result = clean
+      clean
   else:
-    result = ""
+    ""
 
-template user*(): string =
-  $getpwuid(getuid()).pw_name
+proc user*(): string =
+  result = $getpwuid(getuid()).pw_name
 
-proc host*(): string {.inline.} =
-  const size = 64
-  var s = cstring(newString(size))
-  result = $s.gethostname(size)
+proc host*(): string =
+  const size = 64 
+  result = newString(size)
+  discard gethostname(cstring(result), size)
 
-template uidsymbol*(root, user: string): string =
-  if unlikely(getuid() == 0): root else: user
+proc uidsymbol*(root, user: string): string =
+  result = if getuid() == 0: root else: user
 
-func returnCondition*(ok: string, ng: string, delimiter = "."): string {.inline.} =
+func returnCondition*(ok: string, ng: string, delimiter = "."): string = 
   result = fmt"%(?{delimiter}{ok}{delimiter}{ng})"
 
-template returnCondition*(ok: proc(): string, ng: proc(): string, delimiter = "."): string =
-  returnCondition(ok = ok(), ng = ng(), delimiter = delimiter)
-
-func echoc*(s: cstring) {.importc: "printf", header: "<stdio.h>".} ## Fast pure C echo,uses cstring.
```

Comment:
This is again not needed at all - Nim is not so slow that you would actually want to care about `echo` performance in most cases.